packages/remix-dev/compiler/compilerServer.ts

```diff
@@ -44,7 +44,9 @@ const createEsbuildConfig = (
   let isCloudflareRuntime = ["cloudflare-pages", "cloudflare-workers"].includes(
     config.serverBuildTarget ?? ""
   );
-  let isDenoRuntime = config.serverBuildTarget === "deno";
+  let isDenoRuntime = config.serverBuildTarget
+    ? ["deno", "netlify-edge"].includes(config.serverBuildTarget)
+    : false;
```

Comment:
If `config.serverBuildTarget` is `undefined`, `includes` will resolve to `false` as well

```suggestion
  let isDenoRuntime = ["deno", "netlify-edge"].includes(config.serverBuildTarget);
```

Reply:
TypeScript will complain because `Array.prototype.includes`'s type for the parameter does not allow `undefined`. I could put it to `let isDenoRuntime = ["deno", "netlify-edge"].includes(config.serverBuildTarget!)` though.

Reply:
I would prefer using `!` in that case tbh